AEPAssurance/Source/AssuranceStateManager.swift

```diff
@@ -0,0 +1,215 @@
+
+    /// environment property defines the current environment to which the assurance session is connecting.
+    /// This value should not be confused with launch environment.
+    /// This value denotes the environment of Griffon services and UI running to which connection is made.
+    /// This value is obtained from as a query parameter of deeplink URL.
+    var environment: AssuranceEnvironment {
+        get {
+            AssuranceEnvironment.init(envString: datastore.getString(key: AssuranceConstants.DataStoreKeys.ENVIRONMENT) ?? DEFAULT_ENVIRONMENT.rawValue)
+        }
+        set {
+            datastore.set(key: AssuranceConstants.DataStoreKeys.ENVIRONMENT, value: newValue.rawValue)
+        }
+    }
+
+    /// clientID is an identifier to uniquely identify the connected device to an Assurance Session.
+    /// This is a string representation of a UUID. This ID is required for client → server communications.
+    /// A clientID is generated at the client and is persisted throughout the lifecycle of the app.
+    lazy var clientID: String = {
+        // return with clientId, if it is already available in persistence
+        if let persistedClientID = datastore.getString(key: AssuranceConstants.DataStoreKeys.CLIENT_ID) {
+            return persistedClientID
+        }
+
+        // If not generate a new clientId
+        let newClientID = UUID().uuidString
+        datastore.set(key: AssuranceConstants.DataStoreKeys.CLIENT_ID, value: newClientID)
+        return newClientID
+
+    }()
+
+    /// property representing the webSocket URL of the ongoing Assurance session
+    /// A valid value on this property represents that an assurance session is currently running.
+    /// A nil value on this property represents there is no ongoing assurance session.
+    var connectedWebSocketURL: String? {
+        get {
+            datastore.getString(key: AssuranceConstants.DataStoreKeys.SOCKETURL)
+        }
+        set {
+            if let newValue = newValue {
+                datastore.set(key: AssuranceConstants.DataStoreKeys.SOCKETURL, value: newValue)
+            } else {
+                datastore.remove(key: AssuranceConstants.DataStoreKeys.SOCKETURL)
+            }
+        }
+    }
+
+    let runtime: ExtensionRuntime
+    init(_ runtime: ExtensionRuntime) {
+        self.runtime = runtime
+    }
+
+    /// Call this function to create a new shared state for Assurance
+    /// Important - An empty shared state is created if sessionId is not available
+    func shareAssuranceState() {
+        runtime.createSharedState(data: getSharedStateData() ?? [:], event: nil)
+    }
+
+    /// Call this function to empty the latest Assurance shared state
+    func clearAssuranceState() {
+        runtime.createSharedState(data: [:], event: nil)
+    }
+
+    /// Returns an Array of `AssuranceEvent`s containing regular and XDM shared state details of all the registered extensions.
```

Comment:
would like to see this comment explain behavior when the event hub status is not set or is empty